src/connector/src/parser/csv_parser.rs

```diff
@@ -76,7 +77,15 @@ impl CsvParser {
     fn parse_string(dtype: &DataType, v: String) -> AccessResult {
         let v = match dtype {
             // mysql use tinyint to represent boolean
-            DataType::Boolean => (parse!(v, i16)? != 0).into(),
+            DataType::Boolean => {
+                str_to_bool(&v)
+                    .map(ScalarImpl::Bool)
+                    .map_err(|_| AccessError::TypeError {
```

Comment:
`map` + `map_err` is just `match`  😂

Reply:
Functional programming invaded the brain

Reply:
It seems `map` + `map_err` can save 1 line of code.